### PR TITLE
Preparation for component-library button

### PR DESCRIPTION
**Pull request by ValkovMirec**
WIP started to replacing buttons on Bratislava

I have a question about that effective way of navigation trough routes containing components we are working on. I've already asked it on slack with no answer. 

blocked: component-library is not on NPM

Question:

Este by som potreboval poradit ohladom [bratislava.sk](http://bratislava.sk/). Kedze sa tam generuju cesty, sekcie a vsetko spada cez slug. Aky je najlepsi sposob navigacie po stranke ked chcem otestovat, ako sa chova komponent v urcitej sekcii. Mate na to nejaky efektivny sposob?

bratislava/component-library#4

### CODE REVIEW

File: next/components/forms/info-components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import HelpIcon from '@assets/images/forms/icon-help.svg'
+// import Button from '@components/forms/simple-components/Button'
```
Comment (radoslavzeman) [inline next/components/forms/info-components/Tooltip/Tooltip.tsx:2]: Please don't forget to clean up unused imports :)

File: next/components/forms/simple-components/ClipboardCopy.tsx [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
   const [_, copy] = useCopyToClipboard()
   return (
-    // TODO replace by Button component
-    <button type="button" onClick={() => copy(copyText)}>
-      <CopyIcon />
-    </button>
+    <Button
+      variant="icon-wrapped"
+      icon={<CopyIcon />}
+      aria-label="Copy to clipboard"
```
Comment (radoslavzeman) [inline next/components/forms/simple-components/ClipboardCopy.tsx:11]: Please add translation.

File: next/components/forms/simple-components/MLink.tsx [hunk 3]
```diff
@@ -24,7 +24,7 @@ const MLink = forwardRef<HTMLAnchorElement, LinkProps>(
     const plausible = usePlausible()
 
     const styles = twMerge(
-      cx('underline-offset-2', {
```
Comment (radoslavzeman) [inline next/components/forms/simple-components/MLink.tsx:27]: Why this change?

File: next/components/forms/widget-components/Modals/Modal.tsx [hunk 4]
```diff
@@ -104,6 +104,10 @@ const ModalFooter = ({
       >
         <span aria-hidden="true">{cancelLabel}</span>
       </button>
+      MODALITO??
+      <Button variant="black-solid" size="small" onPress={onClose}>
+        <span aria-hidden="true">{cancelLabel}</span>
+      </Button>
```
Comment (radoslavzeman) [inline next/components/forms/widget-components/Modals/Modal.tsx:110]: Why there is span inside the Button please? It should not be needed.

File: next/components/organisms/NavBar/NavMenu/MobileNavMenuContent.tsx [hunk 5]
```diff
@@ -30,14 +31,13 @@ const MobileNavMenuContent = ({ menuItem, colorStyle }: NavMenuContentProps) =>
...
         <li className="w-full">
-          <button
-            type="button"
-            onClick={() => setMenuValue('')}
+          <Button
+            onPress={() => setMenuValue('')}
             className="-my-2 flex w-full items-center gap-2 py-2"
             aria-label={t('aria.backTo', { backTo: parentLabel })}
           >
             <ChevronLeftIcon aria-hidden /> {parentLabel}
```
Comment (radoslavzeman) [inline next/components/organisms/NavBar/NavMenu/MobileNavMenuContent.tsx:39]: Could this icon be used as `startIcon`?

File: next/components/ui/Breadcrumbs/DesktopBreadcrumbs.tsx [hunk 6]
```diff
@@ -21,9 +26,13 @@ const DesktopBreadcrumbs = ({ breadcrumbs }: BreadcrumbsProps) => {
...
             {breadcrumb.path && !isLast ? (
-              <MLink href={breadcrumb.path} variant="underlined">
-                {breadcrumb.title}
-              </MLink>
+              <Button
+                href={breadcrumb.path}
+                variant="unstyled"
+                className="w-full cursor-pointer outline-none"
+              >
+                {breadcrumb.title}{' '}
```
Comment (radoslavzeman) [inline next/components/ui/Breadcrumbs/DesktopBreadcrumbs.tsx:34]: Please remove unwanted space `{' '}`

File: next/components/ui/Breadcrumbs/DesktopBreadcrumbs.tsx [hunk 7]
```diff
@@ -21,9 +26,13 @@ const DesktopBreadcrumbs = ({ breadcrumbs }: BreadcrumbsProps) => {
           <li className="text-small flex gap-1 font-medium" key={index}>
             <BreadcrumbsChevronIcon className="shrink-0" />
             {breadcrumb.path && !isLast ? (
-              <MLink href={breadcrumb.path} variant="underlined">
-                {breadcrumb.title}
-              </MLink>
+              <Button
+                href={breadcrumb.path}
+                variant="unstyled"
+                className="w-full cursor-pointer outline-none"
```
Comment (radoslavzeman) [inline next/components/ui/Breadcrumbs/DesktopBreadcrumbs.tsx:32]: `cursor-pointer` should not be used on random plases, this should be handled in main component.

Never disable `outline-none` if you don't provide other focus style.

File: next/components/ui/Breadcrumbs/DesktopBreadcrumbs.tsx [hunk 8]
```diff
@@ -21,9 +26,13 @@ const DesktopBreadcrumbs = ({ breadcrumbs }: BreadcrumbsProps) => {
           <li className="text-small flex gap-1 font-medium" key={index}>
             <BreadcrumbsChevronIcon className="shrink-0" />
             {breadcrumb.path && !isLast ? (
-              <MLink href={breadcrumb.path} variant="underlined">
-                {breadcrumb.title}
-              </MLink>
```
Comment (radoslavzeman) [inline next/components/ui/Breadcrumbs/DesktopBreadcrumbs.tsx:26]: But TBH, I don't see a reason to use a Button here, MLink is completely okay, isn't it?

File: next/components/ui/Breadcrumbs/MobileBreadcrumbs.tsx [hunk 9]
```diff
@@ -23,20 +23,24 @@ const MobileBreadcrumbs = ({ breadcrumbs }: BreadcrumbsProps) => {
     <div className="relative">
       <div className="flex justify-between">
         <div className="text-small -mx-4 flex items-center gap-2 font-medium">
-          {/* TODO: Convert to shared link/button component. Detect if user can actually go back. */}
-          <button
-            type="button"
-            className="flex items-center gap-1 py-3 pl-4 underline hover:text-gray-600"
-            onClick={goBack}
+          <Button
```
Comment (radoslavzeman) [inline next/components/ui/Breadcrumbs/MobileBreadcrumbs.tsx:26]: Please provide a screenshot before and after the change.

File: next/components/ui/Breadcrumbs/MobileBreadcrumbs.tsx [hunk 10]
```diff
@@ -23,20 +23,24 @@ const MobileBreadcrumbs = ({ breadcrumbs }: BreadcrumbsProps) => {
...
+            startIcon={<BreadcrumbsChevronIcon className="shrink-0 rotate-180" />}
           >
-            <BreadcrumbsChevronIcon className="shrink-0 rotate-180" />
-            <span>{t('back')}</span>
-          </button>
+            {t('back')}
+          </Button>
           <div className="h-4 w-px bg-gray-300" />
           {last.path ? (
-            <MLink href={last.path} variant="underlined" aria-hidden className="truncate py-3 pr-4">
```
Comment (radoslavzeman) [inline next/components/ui/Breadcrumbs/MobileBreadcrumbs.tsx:37]: I think MLink is okay here.

File: next/components/ui/Breadcrumbs/MobileBreadcrumbs.tsx [hunk 11]
```diff
@@ -54,10 +58,10 @@ const MobileBreadcrumbs = ({ breadcrumbs }: BreadcrumbsProps) => {
               // eslint-disable-next-line react/no-array-index-key
               <li className="text-small font-medium" key={index}>
                 {breadcrumb.path ? (
-                  <MLink href={breadcrumb.path} variant="underlined" className="flex gap-1">
+                  <Button href={breadcrumb.path} className="flex gap-1">
```
Comment (radoslavzeman) [inline next/components/ui/Breadcrumbs/MobileBreadcrumbs.tsx:61]: MLink is kay here.
Moreover, you didn't provide underline style for the Button.

File: next/components/ui/RegulationListItem/RegulationListItem.tsx
Comment (radoslavzeman) [inline next/components/ui/RegulationListItem/RegulationListItem.tsx:1]: Oh yes, I'm very glad that we can use our Button finally :)

File: next/pages/styleguide.tsx [hunk 12]
```diff
@@ -26,40 +30,58 @@ import TextAreaFieldShowCase from '../components/styleguide/showcases/TextAreaFi
 import ToggleShowCase from '../components/styleguide/showcases/ToggleShowCase'
 import StyleGuideWrapper from '../components/styleguide/StyleGuideWrapper'
 
+const LinkWrapper = ({
+  href,
+  children,
+  className,
```
Comment (radoslavzeman) [inline next/pages/styleguide.tsx:36]: What about other other link props?

File: next/pages/styleguide.tsx [hunk 13]
```diff
@@ -26,40 +30,58 @@ import TextAreaFieldShowCase from '../components/styleguide/showcases/TextAreaFi
...
-      <CheckboxGroupShowCase />
-      <AccordionShowcase />
-      <BannerShowCase />
-      <ServiceCardShowCase />
-      <EventCardShowcase />
-      <CategoryCardShowcase />
-      <BlogPostCardShowcase />
-      <HomepageHorizontalCardShowcase />
-      <ContactsShowcase />
+    <ComponentLibraryEnvironmentContext.Provider value={{ Link: LinkWrapper }}>
```
Comment (radoslavzeman) [inline next/pages/styleguide.tsx:55]: Don't we need provider also on the whole app, not just styleguide?

**Review by radoslavzeman — CHANGES_REQUESTED** [review]
Please address TS errors and all comments.

File: next/components/forms/info-components/Tooltip/Tooltip.tsx (hunk 1 above)
Comment (ValkovMirec) [reply next/components/forms/info-components/Tooltip/Tooltip.tsx:2]: yes, about that TS errors. I forgot to remove imported component library in styleguide. PR in component-library with code required for it is not merged yet so we cant really import it, neither can we use ContextProvider in any place for now. It was tested locally with yalc. As for other issues, I'm looking trough it right now.

File: next/components/forms/widget-components/Modals/Modal.tsx (hunk 4 above)
Comment (ValkovMirec) [reply next/components/forms/widget-components/Modals/Modal.tsx:110]: I must have left it there when using new button. Will delete. But, I've noticed that this modal is not used. It's imported to AccountModal but this modal isn't imported anywhere. Only modal made from RAC components is used  Should we keep it?

File: next/components/organisms/NavBar/NavMenu/MobileNavMenuContent.tsx (hunk 5 above)
Comment (ValkovMirec) [reply next/components/organisms/NavBar/NavMenu/MobileNavMenuContent.tsx:39]: yes.. that one is one of the first ones I did. forgot about "icon" props

File: next/components/ui/Breadcrumbs/DesktopBreadcrumbs.tsx (hunk 8 above)
Comment (ValkovMirec) [reply next/components/ui/Breadcrumbs/DesktopBreadcrumbs.tsx:26]: Well It would be sufficient. But since Button contains and uses MLink when handling href. Isn't it ok to handle these links trough Button with href?

File: next/components/ui/Breadcrumbs/MobileBreadcrumbs.tsx (hunk 9 above)
Comment (ValkovMirec) [reply next/components/ui/Breadcrumbs/MobileBreadcrumbs.tsx:26]: Screenshot 2023-12-14 at 11 10 31
Screenshot 2023-12-14 at 11 10 52

File: next/components/forms/widget-components/Modals/Modal.tsx (hunk 4 above)
Comment (radoslavzeman) [reply next/components/forms/widget-components/Modals/Modal.tsx:110]: You can remove unused file.

File: next/components/ui/Breadcrumbs/DesktopBreadcrumbs.tsx (hunk 8 above)
Comment (radoslavzeman) [reply next/components/ui/Breadcrumbs/DesktopBreadcrumbs.tsx:26]: No, if it would change the design even slightly, please use MLink.

File: next/components/forms/simple-components/MLink.tsx (hunk 3 above)
Comment (ValkovMirec) [reply next/components/forms/simple-components/MLink.tsx:27]: that was to allign offset of Button link variant and MLink underline. But after I've looked trough it, it's better to set it in Button variants then to remove it from MLink

File: next/components/forms/simple-components/Button.tsx [hunk 14]
```diff
@@ -182,8 +182,9 @@ const Button = forwardRef<HTMLAnchorElement | HTMLButtonElement, PolymorphicProp
                 'text-negative-700 pressed:bg-negative-200 pressed:text-negative-800':
                   variant === 'negative-plain',
 
-                'text-category-700 pressed:text-category-800': variant === 'category-link',
-                'text-gray-700 pressed:text-gray-800': variant === 'black-link',
+                'underline-offset-2 text-category-700 pressed:text-category-800':
+                  variant === 'category-link',
+                'underline-offset-2 text-gray-700 pressed:text-gray-800': variant === 'black-link',
```
Comment (radoslavzeman) [inline next/components/forms/simple-components/Button.tsx:187]: I would add `underline-offset-2`  to `isLinkVariant` condition (line 137)

File: next/components/forms/simple-components/ClipboardCopy.tsx [hunk 15]
```diff
@@ -1,13 +1,19 @@
 import { CopyIcon } from '@assets/ui-icons'
+import Button from '@components/forms/simple-components/Button'
+import { useTranslations } from 'next-intl'
 import { useCopyToClipboard } from 'usehooks-ts'
 
 const ClipboardCopy = ({ copyText }: { copyText: string }) => {
-  const [_, copy] = useCopyToClipboard()
+  // eslint-disable-next-line no-unused-vars
```
Comment (radoslavzeman) [inline next/components/forms/simple-components/ClipboardCopy.tsx:7]: Not needed probably.

**Review by radoslavzeman — APPROVED** [review]
I think it's ready to merge when you address these last comments :)
